.github/workflows/publish.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish to NPM
+on:
+  release:
+    types: [created]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+
+      - name: Install pnpm
+        id: install-pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
+          run_install: false
+
+      - name: Setup Node
+        id: setup-node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'pnpm'
+
+      - name: Install dependencies and build 🔧
```

Comment:
Can we break out the dependency installation from the build? In particular, I believe we should use the `--frozen-lockfile` option, per the example linked above. I'm generally uncomfortable with the builders being able to silently update dependencies vs. what was tested by developers on their machines. I'd rather it select dependencies predictably, from the preconfigured lockfile.